app/models/menu/menu.rb

```diff
@@ -618,13 +618,14 @@ def hmis_admin_menu
   end
 
   def hmis_menu
-    # NOTE: eventually we should use the data source name here, maybe concatenated with HMIS, but currently that would be HMIS HMIS
-    GrdaWarehouse::DataSource.hmis.distinct.map do |hmis_ds|
+    hmis_data_sources = GrdaWarehouse::DataSource.hmis
+    hmis_data_sources.map do |hmis_ds|
+      default_link_text = hmis_data_sources.size == 1 ? 'Open HMIS' : "Open #{hmis_ds.short_name}"
       Menu::Item.new(
         user: user,
-        visible: ->(user) { user.any_hmis_access? },
+        visible: ->(user) { user.can_access_hmis_data_source?(hmis_ds.id) },
         path: "//#{hmis_ds.hmis}",
-        title: Translation.translate('Open HMIS'),
+        title: Translation.translate(default_link_text),
```

Comment:
Full link text is customizable, whether there is 1 hmis or multiple
<img width="188" alt="Screenshot 2024-10-16 at 8 27 05 PM" src="https://github.com/user-attachments/assets/28e56404-5c2d-47f0-9e07-f4bd6d20239e">